modules/compute/variables.tf

```
@@ -0,0 +1,44 @@
+variable "location" {
+  description = "The location where resources will be created"
+  type        = string
+  default     = "uksouth"
+
+}
+
+variable "resource_group_name" {
+  description = "The name of the resource group"
+  type        = string
+  default     = "mate-azure-task-12"
```

`variable "location"` and `variable "resource_group_name"` have defaults (`uksouth`, `mate-azure-task-12`). The task expects these values to be defined at the root level (root variables.tf / terraform.tfvars). Providing defaults inside the module is not strictly wrong, but it can mask configuration mistakes and conflicts with the task's expectation. I recommend removing these defaults from the module and letting the root pass them explicitly (centralized configuration) .